composer/trainer/mosaic_fsdp_utils.py

```diff
@@ -65,6 +64,51 @@
 logger = logging.getLogger(__name__)
 
 
+def set_fsdp_default(fsdp_config: Dict[str, Any]):
+    """Modify fsdp_config to set default values for missing keys."""
+    if 'process_group' in fsdp_config:
+        warnings.warn(VersionedDeprecationWarning('process_group is deprecated. Please specify `data_parallel_shard_degree` and `data_parallel_replicate_degree` instead.', remove_version='0.24.0'))
+
+    if 'device_mesh' in fsdp_config:
+        warnings.warn(VersionedDeprecationWarning('device_mesh is deprecated. Please specify `data_parallel_shard_degree` and `data_parallel_replicate_degree` instead.', remove_version='0.24.0'))
+        if 'data_parallel_shard_degree' in fsdp_config or 'data_parallel_replicate_degree' in fsdp_config:
+            raise ValueError('Cannot specify both `device_mesh` and `data_parallel_shard_degree` or `data_parallel_replicate_degree`. Please remove `device_mesh`.')
+        device_mesh = fsdp_config.pop('device_mesh')
+        fsdp_config['data_parallel_shard_degree'] = device_mesh[0]
+        if len(device_mesh) > 1:
+            fsdp_config['data_parallel_replicate_degree'] = device_mesh[1]
+
+    fsdp_config.setdefault('activation_checkpointing', False)
```

Comment:
these will be handled by a dataclass in a future PR right

Reply:
Yep!